### PR TITLE
Build integration test in PRs originating from repo

### PR DESCRIPTION
**Pull request by RustedBones**
Catch integration test compilation issue inside PR by compiling the `integration/Test` when PR ogiginate from this repo (compilation requires acccess to `data-integration-test` GCP project)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -82,12 +82,20 @@ jobs:
         if: matrix.java == 'corretto@21' && steps.setup-java-corretto-21.outputs.cache-hit == 'false'
         run: sbt +update
 
+      - name: gcloud auth
+        if: (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
```
Comment (clairemcginty) [inline .github/workflows/ci.yml:86]: we could also just check if `!pull_request.head.repo.fork` - wdyt?

Comment (RustedBones) [reply .github/workflows/ci.yml:86]: Thanks!

[the remark this answers is not in the document]
Comment (RustedBones) [reply .github/workflows/ci.yml:86]: integration test on main build for instance

**Review by clairemcginty — APPROVED** [review]

**Comment by RustedBones** [comment]
@clairemcginty I changed the strategy:
- `integration` is not part of the aggregated projects
- test in integration are skipped by default and must be manually enabled. Following log appears if not enabled when running `integration/test`:
```
sbt:scio> integration/test
[warn] integration/test are skipped.
[warn] Run 'set integration/test/skip := false' to run them
```
- `site` and `scio-examples` are only build/tested if `gcould auth` is ran on CI. Those module require a gcp project. CI builds on forks do not expose repository secret to avoid attacks.

**Review by clairemcginty — APPROVED** [review]

**Pull request merged**